Sources/MongoSwift/Operations/ListCollectionsOperation.swift

```diff
@@ -0,0 +1,110 @@
+import mongoc
+
+/// Describes the type of data store returned when executing `listCollections`
+public enum CollectionType: RawRepresentable, Codable {
+    /// Specifies that the data store returned is a collection.
+    case collection
+    /// Specifies that the data store returned is a view.
+    case view
+    /// For an unknown value. For forwards compatibility, no error will be thrown when an unknown value is provided.
+    case other(String)
+
+    public var rawValue: String {
+        switch self {
+        case .collection:
+            return "collection"
+        case .view:
+            return "view"
+        case .other(let v):
+            return v
+        }
+    }
+
+    public init(rawValue: String) {
+        switch rawValue {
+        case "collection":
+            self = .collection
+        case "view":
+            self = .view
+        default:
+            self = .other(rawValue)
+        }
+    }
+}
+
+/// Specifications of a collection returned when executing `listCollections`.
+public struct CollectionSpecification: Codable {
+    /// The name of the collection.
+    public let name: String
+
+    /// The type of data store returned.
+    public let type: CollectionType
+
+    /// Options that were used when creating this collection.
+    public let options: CreateCollectionOptions?
+}
+
+/// Options to use when executing a `listCollections` command on a `MongoDatabase`.
+public struct ListCollectionsOptions: Encodable {
+    /// The batchSize for the returned cursor.
+    public var batchSize: Int?
+
+    /// Convenience initializer allowing any/all parameters to be omitted or optional
+    public init(batchSize: Int? = nil) {
+        self.batchSize = batchSize
+    }
+}
+
+/// Internal intermediate result of a ListCollections command.
+internal enum ListCollectionsResults {
+    /// Includes the name, type, and creation options of each collection.
+    case specs(MongoCursor<CollectionSpecification>)
+
+    /// Only includes the names.
+    case names([String])
+}
+
+/// An operation corresponding to a "listCollections" command on a database.
+internal struct ListCollectionsOperation: Operation {
+    private let database: MongoDatabase
+    private let filter: Document?
+    private let options: ListCollectionsOptions?
+    private let nameOnly: Bool?
+
+    internal init(database: MongoDatabase, filter: Document? = nil, options: ListCollectionsOptions?, nameOnly: Bool?) {
```

Comment:
nit: does `filter` need a default value? I think we are always passing it in whenever we call this initializer